### DISCUSSION TITLE
Bot thinks it's checkmating

### DISCUSSION THREAD

**Discussion by 2tothe4th** [SebLague/Chess-Challenge#185] [Q&A] [answered]
I'm not sure if I'm just crazy but this code
```cs
int checkmateMultiplier = board.IsWhiteToMove ? 1 : -1;

if (board.IsInCheckmate()) 
{
    evaluation = int.MinValue * checkmateMultiplier;
}
```
makes it so when I checkmate the bot, the console prints
```
1rbqkbnr/pppppBp1/8/7p/4P3/1Q5N/PP1P1PPP/nNB3KR b k - 5 9: -2147483648
```
the right number being the evaluation (positive: white is winning, negative: black is winning).
I could just be overlooking something, but I would like some help.

**Comment by WhiteMouse1** [comment, marked as answer]
int.MinValue can retain its negative value even when multiplied by negative number.
To be sure there are no bugs like that, use large numbers as max/min values, i.e 99999 and -99999 correspondingly.
For chess evaluation purposes those numbers should suffice.

I had half of my day destroyed by the same problem.

**Reply by 2tothe4th** [reply]
This seemed to help the bot find checkmate. Thanks!
